Katana/Store.swift

```diff
@@ -441,20 +445,21 @@ open class Store<S: State, D: SideEffectDependencyContainer>: PartialStore<S> {
   fileprivate func nonPromisableDispatch(_ dispatchable: Dispatchable) {
     self.anyDispatch(dispatchable)
   }
-  
-  /// This method is used to free all the references, since the store has some reference loops
-  /// going on with various entities. This meant to be called only when you are no longer using
-  /// the store and you want it to be garbage collected. Keep in mind this is not a safe method,
-  /// if someone else still has a reference to the store and tries to dispatch something or
-  /// to access the state/dependencies the application will crash.
-  func __unsafe_releaseReferences() {
-    self.isReady = false
-    self.dependencies = nil
-    self.initializedInterceptors = []
-    self.sideEffectContext = nil
-    self.stateUpdaterQueue = nil
-    self.sideEffectQueue = nil
-    self.mainAsyncProvider = nil
+  /// The dependencies used to initialize katana
+  public struct Configuration {
+    /// AsyncProvider used to run the state initializer
+    let stateInitializerAsyncProvider: AsyncProvider
```

Comment:
why do you decide to split them? Is there a specific reason?

Reply:
as discussed with @bolismauro, they are used to do different things. so it's a bit cleaner to ask for the appropriate async provider for each case